packages/twenty-server/src/workspace/workspace-query-builder/factories/find-duplicates-query.factory.ts

```diff
@@ -0,0 +1,163 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import isEmpty from 'lodash.isempty';
+
+import { WorkspaceQueryBuilderOptions } from 'src/workspace/workspace-query-builder/interfaces/workspace-query-builder-options.interface';
+import { RecordFilter } from 'src/workspace/workspace-query-builder/interfaces/record.interface';
+import { FindDuplicatesResolverArgs } from 'src/workspace/workspace-resolver-builder/interfaces/workspace-resolvers-builder.interface';
+import { ObjectMetadataInterface } from 'src/metadata/field-metadata/interfaces/object-metadata.interface';
+import { PGGraphQLResult } from 'src/workspace/workspace-query-runner/interfaces/pg-graphql.interface';
+
+import { computeObjectTargetTable } from 'src/workspace/utils/compute-object-target-table.util';
+import { stringifyWithoutKeyQuote } from 'src/workspace/workspace-query-builder/utils/stringify-without-key-quote.util';
+import { ArgsAliasFactory } from 'src/workspace/workspace-query-builder/factories/args-alias.factory';
+
+import { FieldsStringFactory } from './fields-string.factory';
+
+type DuplicateCriteria = {
```

Comment:
```suggestion
type RecordDuplicateCriteria = {
```

Reply:
BTW: As we're going to move the duplicate criteria outside we should move this type into the `interfaces` folder and instead use `interface` over `type`.